src/app/DataVersionFilter.h

```diff
@@ -28,13 +29,22 @@ struct DataVersionFilter
         mClusterId(aClusterId), mDataVersion(aDataVersion), mEndpointId(aEndpointId)
     {}
 
+    DataVersionFilter(EndpointId aEndpointId, ClusterId aClusterId, Optional<DataVersion> aDataVersion) :
```

Comment:
We need this because we are initializing these things from a `ConcreteDataAttributePath`?

Seems to me we should have a `ConcreteDataWriteAttributePath` (optional dataversion) and a `ConcreteDataReadAttributePath` (non-optional data version)...

Reply:
yes, in OnUpdateDataVersionFilterList, we need init DataVersionFilter dataVersionFilter(path1->mEndpointId, path1->mClusterId, path1->mDataVersion) using ConcreteDataAttributePathWithSize.

https://github.com/project-chip/connectedhomeip/issues/15017